str/data_frame_creator.py

```diff
@@ -0,0 +1,222 @@
+                'e_allele_2',
+                'e_repci',
+                'e_adsp',
+                'e_adfl',
+                'e_adir',
+                'e_lc',
+            ]
+        )
+        + '\n'
+    )
+
+    for file in files:
+        if isinstance(file, GSPath):
+            file.copy(file.name)
+            file = file.name
+        sample_id = str(VCFReader(file).samples[0])
+        for variant in VCFReader(file):
+            chr = str(variant.CHROM)
+            start = str(variant.POS)
+            e_qual = str(variant.FILTER)
+            end = str(variant.INFO.get('END'))
+            repeat_units_in_ref = str(variant.INFO.get('REF'))
+            ref_sequence_length = str(variant.INFO.get('RL'))
+            motif = str(variant.INFO.get('RU'))
+            e_gt = f'{variant.genotypes[0][0]}/{variant.genotypes[0][1]}'
+            e_so = str(variant.format('SO')[0])
+            e_allele_1 = str(variant.format('REPCN')[0].split('/')[0])
+            e_allele_2 = str(variant.format('REPCN')[0].split('/')[1])
+            e_repci = str(variant.format('REPCI')[0])
+            e_adsp = str(variant.format('ADSP')[0])
+            e_adfl = str(variant.format('ADFL')[0])
+            e_adir = str(variant.format('ADIR')[0])
+            e_lc = str(variant.format('LC')[0][0])
+            csv = csv + (
+                ','.join(
+                    [
+                        sample_id,
+                        chr,
+                        start,
+                        e_qual,
+                        end,
+                        repeat_units_in_ref,
+                        ref_sequence_length,
+                        motif,
+                        e_gt,
+                        e_so,
+                        e_allele_1,
+                        e_allele_2,
+                        e_repci,
+                        e_adsp,
+                        e_adfl,
+                        e_adir,
+                        e_lc,
+                    ]
+                )
+                + '\n'
+            )
+    return csv
+
+
+def gangstr_tsv_writer(input_dir):
+    """Creates a TSV file containing dataframe of merged GangSTR VCFs"""
+    files = to_path(input_dir).glob('*.vcf')
+    tsv = (
+        '\t'.join(
+            [
+                'sample_id',
+                'chr',
+                'start',
+                'g_gt',
+                'g_dp',
+                'g_q',
+                'g_repci',
+                'g_rc',
+                'g_enclreads',
+                'g_flnkreads',
+                'g_ml',
+                'g_ins',
+                'g_stderr',
+                'g_allele_1',
+                'g_allele_2',
+                'g_qexp',
+            ]
+        )
+        + '\n'
+    )
+    for file in files:
+        if isinstance(file, GSPath):
+            file.copy(file.name)
+            file = file.name
+        sample_id = str(VCFReader(file).samples[0])
+        for variant in VCFReader(file):
```

Comment:
```suggestion
        reader = VCFReader(file)
        sample_id = str(reader.samples[0])
        for variant in reader:
```

Not a big performance hit, but when you create a VCFReader it absorbs all the Header data, Samples, etc. Here you're creating one populated object, throwing it away, then repeating with the second object.